curator-recipes/src/test/java/org/apache/curator/framework/recipes/shared/TestSharedCount.java

```diff
@@ -381,12 +385,122 @@ public void processResult(CuratorFramework client, CuratorEvent event) throws Ex
             }).forPath("/count");
             flushDone.await(5, TimeUnit.SECONDS);
 
-            Assert.assertEquals(2, numChangeEvents.get());
+            // CURATOR-311: when a Curator client's state became RECONNECTED, countHasChanged method is called back
+            // because the Curator client calls readValueAndNotifyListenersInBackground in SharedValue#ConnectionStateListener#stateChanged.
+            Assert.assertEquals(numChangeEvents.get(), 3);
```

Comment:
Minor nits: this line also fixes wrong order of the argument of Assert.assertEquals: assertEquals(actual, expected)  is correct one.

To confirm the reason why I changed numChangeEvents' count, please check the comment in the test code.